### PR TITLE
doc: Add performance tuning section

### PR DESCRIPTION
**Pull request by hermanschaaf**
I started this PR mainly as a way to document the new wildcard options, but ultimately updated a few things relating to the CloudQuery docs:
 - added brief notes about wildcards to the source plugin reference sections for `tables` and `skip_tables`
 - placed detailed information about wildcards in a `Performance tuning` page under `Advanced Topics`. The idea is that we will expand this page over time.
 - updated the docs relating to `concurrency` (`resource_concurrency` and `table_concurrency` were deprecated)
 - some other misc fixes

### CODE REVIEW

File: website/pages/docs/advanced-topics/performance-tuning.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+title: Performance Tuning
+---
+
+# Performance tuning
```
Comment (disq) [inline website/pages/docs/advanced-topics/performance-tuning.md:5]:
```suggestion
# Performance Tuning
```

**Review by disq — APPROVED** [review]
LGTM with a single nit

File: website/pages/docs/advanced-topics/performance-tuning.md (hunk 1 above)
Comment (hermanschaaf) [reply website/pages/docs/advanced-topics/performance-tuning.md]: Thanks!

File: website/pages/docs/reference/cli/_meta.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
   "cloudquery": "cloudquery",
-  "cloudquery_sync": "cloudquery sync"
+  "cloudquery_sync": "cloudquery sync",
+  "cloudquery_migrate": "cloudquery migrate"
```
Comment (erezrokah) [inline website/pages/docs/reference/cli/_meta.json:4]: Nice, maybe we can add something to our CLI Makefile https://github.com/cloudquery/cloudquery/blob/75162dd372e17b90a387a552d63f61d4c177131b/cli/Makefile#L13 to generate the `_meta.json` file (unrelated to this PR)

**Review by erezrokah — APPROVED** [review]
Looks great 🚀 Nothing blocking

File: website/pages/docs/advanced-topics/performance-tuning.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+</Callout>
+
+## Improving Performance by Skipping Relations
+
+Some tables require many API calls to sync. This is especially true of tables that depend on other tables, because often multiple API calls need to be made for every row in the parent table. This can lead to thousands of API calls, increasing the time it takes to sync. If you know that some child tables are not strictly necessary, you can improve sync performance by skipping them with the `skip_tables` setting.
+
+Let's say we have three tables: `A`, `B` and `C`. `A` is the top-level table. `B` depends on `A`, and `C` depends on `B`:
+
+```text
+A <- B <- C
```
[the remark this answers is not in the document]
Comment (hermanschaaf) [reply website/pages/docs/advanced-topics/performance-tuning.md:44]: Nice, good idea 👍

**Pull request merged**